mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxStreamableServerTransportProvider.java

```diff
@@ -0,0 +1,479 @@
+							.bodyValue(initResult));
+				}
+
+				if (!request.headers().asHttpHeaders().containsKey("mcp-session-id")) {
+					return ServerResponse.badRequest().bodyValue(new McpError("Session ID missing"));
+				}
+
+				String sessionId = request.headers().asHttpHeaders().getFirst("mcp-session-id");
+				McpStreamableServerSession session = sessions.get(sessionId);
+
+				if (session == null) {
+					return ServerResponse.status(HttpStatus.NOT_FOUND)
+						.bodyValue(new McpError("Session not found: " + sessionId));
+				}
+
+				if (message instanceof McpSchema.JSONRPCResponse jsonrpcResponse) {
+					return session.accept(jsonrpcResponse).then(ServerResponse.accepted().build());
+				}
+				else if (message instanceof McpSchema.JSONRPCNotification jsonrpcNotification) {
+					return session.accept(jsonrpcNotification).then(ServerResponse.accepted().build());
+				}
+				else if (message instanceof McpSchema.JSONRPCRequest jsonrpcRequest) {
+					return ServerResponse.ok()
+						.contentType(MediaType.TEXT_EVENT_STREAM)
+						.body(Flux.<ServerSentEvent<?>>create(sink -> {
+							WebFluxStreamableMcpSessionTransport st = new WebFluxStreamableMcpSessionTransport(sink);
+							Mono<Void> stream = session.responseStream(jsonrpcRequest, st);
+							Disposable streamSubscription = stream.doOnError(err -> sink.error(err))
+								.contextWrite(sink.contextView())
+								.subscribe();
+							sink.onCancel(streamSubscription);
+						}), ServerSentEvent.class);
+				}
+				else {
+					return ServerResponse.status(HttpStatus.INTERNAL_SERVER_ERROR)
+						.bodyValue(new McpError("Unknown message type"));
+				}
+			}
+			catch (IllegalArgumentException | IOException e) {
+				logger.error("Failed to deserialize message: {}", e.getMessage());
+				return ServerResponse.badRequest().bodyValue(new McpError("Invalid message format"));
+			}
+		})
+			.switchIfEmpty(ServerResponse.badRequest().build())
+			.contextWrite(ctx -> ctx.put(McpTransportContext.KEY, transportContext));
+	}
+
+	private Mono<ServerResponse> handleDelete(ServerRequest request) {
+		if (isClosing) {
+			return ServerResponse.status(HttpStatus.SERVICE_UNAVAILABLE).bodyValue("Server is shutting down");
+		}
+
+		McpTransportContext transportContext = this.contextExtractor.apply(request);
+
+		return Mono.defer(() -> {
+			if (!request.headers().asHttpHeaders().containsKey("mcp-session-id")) {
+				return ServerResponse.badRequest().build(); // TODO: say we need a session
+				// id
+			}
+
+			// TODO: The user can configure whether deletions are permitted
+			if (this.disallowDelete) {
+				return ServerResponse.status(HttpStatus.METHOD_NOT_ALLOWED).build();
+			}
+
+			String sessionId = request.headers().asHttpHeaders().getFirst("mcp-session-id");
+
+			McpStreamableServerSession session = this.sessions.get(sessionId);
+
+			if (session == null) {
+				return ServerResponse.notFound().build();
+			}
+
+			return session.delete().then(ServerResponse.ok().build());
```

Comment:
I guess we need to remove the session from the sessions maps as well? 
e.g. `this.sessions.remove(sessionId);`